### PR TITLE
Finished Project 4 Part1

### PR DESCRIPTION
**Pull request by renatix23**
Ready for review

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -40,9 +40,151 @@ send me a DM to check your pull request
...
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    return lhs / rhs;
+}
```
Comment (matkatmusic) [inline main.cpp]: - [x] you forgot the warning that `FloatType::divide` has

File: main.cpp [hunk 2]
```diff
@@ -40,9 +40,151 @@ send me a DM to check your pull request
...
+    std::cout <<"Result of it.add(): " << resultIt << std::endl;
+
+    resultIt = it.substract(20, 4);
+    std::cout <<"Result of it.substract(): " << resultIt << std::endl;
+
+    resultIt = it.multiply(8, 3);
+    std::cout <<"Result of it.multiply(): " << resultIt << std::endl;
+
+    resultIt = it.divide(20, 0);
+    std::cout <<"Result of it.divide(): " << resultIt << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: add a space after the `<<` for all of these.
- [x] improve the readability

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Good!  2 small changes needed

File: main.cpp (hunk 1 above)
Comment (renatix23) [reply main.cpp]: I check it and it's there, the lines you are pointing to are from the Double Type

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**